sakeejsonrpc.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-License-Identifier: GPL-3.0
+import io
+import json
+import os
+import re
+
+import xbmc
+
+
+class JsonRpcApi(object):
+
+    __ADDON_INFO = None
+
+    def __init__(self):
+        """ Initialise the JSON RPC API Implementation. """
+        if JsonRpcApi.__ADDON_INFO is None:
+            JsonRpcApi.__ADDON_INFO = xbmc.get_add_on_info_from_calling_script()
+
+    def handle(self, json_data):
+        """ Handle the JSON RPC Request
+
+        :param obj json_data: The JSON RPC request.
+
+        """
+        (namespace, method_name) = json_data["method"].split('.')
+
+        # Find class
+        try:
+            class_ = getattr(JsonRpcApi, namespace)
```

Comment:
could you name this `class_name`

Reply:
I've used `class_reference`, since namespace is actually the `class_name`.